service/cluster/xtra_db_cluster.go

```diff
@@ -147,14 +167,21 @@ func (s *XtraDBClusterService) CreateXtraDBCluster(ctx context.Context, req *con
 			ComputeResources: computeResources(req.Params.Pxc.ComputeResources),
 			DiskSize:         convertors.BytesToStr(req.Params.Pxc.DiskSize),
 		},
-		ProxySQL: &k8sclient.ProxySQL{
+
+		PMMPublicAddress: req.PmmPublicAddress,
+	}
+	if req.Params.Proxysql != nil {
+		params.ProxySQL = &k8sclient.ProxySQL{
 			Image:            req.Params.Proxysql.Image,
 			ComputeResources: computeResources(req.Params.Proxysql.ComputeResources),
 			DiskSize:         convertors.BytesToStr(req.Params.Proxysql.DiskSize),
-		},
-		PMMPublicAddress: req.PmmPublicAddress,
+		}
+	} else {
```

Comment:
this else branch can be removed using this 

```go
if req.Params.Proxysql != nil {
	params.ProxySQL = &k8sclient.ProxySQL{
		Image:            req.Params.Proxysql.Image,
		ComputeResources: computeResources(req.Params.Proxysql.ComputeResources),
		DiskSize:         convertors.BytesToStr(req.Params.Proxysql.DiskSize),
	}

	err = client.CreateXtraDBCluster(ctx, params)
	if err != nil {
		return nil, status.Error(codes.Internal, err.Error())
    }
	return new(controllerv1beta1.CreateXtraDBClusterResponse), nil
}

params.HAProxy = &k8sclient.HAProxy{
	Image:            req.Params.Haproxy.Image,
	ComputeResources: computeResources(req.Params.Haproxy.ComputeResources),
}
err = client.CreateXtraDBCluster(ctx, params)
if err != nil {
	return nil, status.Error(codes.Internal, err.Error())
}
return new(controllerv1beta1.CreateXtraDBClusterResponse), nil
```

Reply:
this code will be harder to read and goes against makes us duplicate code.

Reply:
@palash25 If it was one line of code we duplicate and `else` block was larger I'd do it. But like this, I think it's not worth it.